shunting-yard-common/src/test/java/com/hotels/shunting/yard/common/messaging/MessageTest.java

```diff
@@ -0,0 +1,130 @@
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload(null);
+    objectUnderTest.timestamp(1515585600000L);
+
+    assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> objectUnderTest.build())
+        .withMessage("Parameter 'payload' is required");
+  }
+
+  @Test
+  public void checkEmpty() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+
+    assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> objectUnderTest.build())
+        .withMessage("Parameter 'table' is required");
+  }
+
+  @Test
+  public void buildOutputNotNull() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+
+    final Message retval = objectUnderTest.build();
+
+    assertThat(retval).isNotNull();
+    assertThat(retval.getDatabase()).isEqualTo("test_db");
+    assertThat(retval.getTable()).isEqualTo("test_table");
+    assertThat(retval.getPayload()).isEqualTo("foo");
+    assertThat(retval.getTimestamp()).isEqualTo(1515585600000L);
+  }
+
+
+  @Test
+  public void getDatabaseOutputNull() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+    final Message message = objectUnderTest.build();
+
+    assertThat(message.getDatabase()).isEqualTo("test_db");
+  }
+
+  @Test
+  public void getPayloadOutputNull() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+    final Message message = objectUnderTest.build();
+
+    assertThat(message.getPayload()).isEqualTo("foo");
+  }
+
+  @Test
+  public void getQualifiedTableNameOutputNotNull() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+    final Message message = objectUnderTest.build();
+
+    assertThat(message.getQualifiedTableName()).isEqualTo("test_db.test_table");
+  }
+
+  @Test
+  public void getTableOutputNull() throws InvocationTargetException {
```

Comment:
why `OutputNull`?

Reply:
because I forgot to refactor some of the test names...